### PR TITLE
Reset signalling state when answer a call after receiving 180/183 with SDP.

### PR DESCRIPTION
**Pull request by cloudwebrtc**
I added state storage to flutter-webrtc.
https://github.com/cloudwebrtc/flutter-webrtc/commit/396af855ac7f13a378c4dcd87856e60dfdfeb7a2
Try to fix the error after receiving 180/183 with SDP.

### CODE REVIEW

File: lib/src/RTCSession.dart [hunk 1]
```diff
@@ -2266,8 +2266,8 @@ class RTCSession extends EventEmitter {
 
       // Be ready for 200 with SDP after a 180/183 with SDP.
       // We created a SDP 'answer' for it, so check the current signaling state.
-      //if (this._connection.signalingState == RTCSignalingState.RTCSignalingStateStable)
-      /*{
+      if (this._connection.signalingState ==
+          RTCSignalingState.RTCSignalingStateStable) {
```
Comment (rlsutton1) [inline lib/src/RTCSession.dart]: changing this line to 

> if (this._connection.signalingState ==
>           RTCSignalingState.RTCSignalingStateHaveLocalOffer)

seems to work.

Note this change also requires a change in flutter-webrtc to make **this._connection.signalingState** accessable.

Comment (rlsutton1) [inline lib/src/RTCSession.dart]: It works if this line is changed to 

> RTCSignalingState.RTCSignalingStateHaveLocalOffer)  {

Comment (cloudwebrtc) [reply lib/src/RTCSession.dart]: It looks wrong. If the status is not RTCSignalingStateHaveLocalOffer after receiving 180/183 with SDP, maybe we should check if the early media is properly connected.

**Comment by ghenry** [comment]
Isn't the design already in JsSIP?

**Comment by cloudwebrtc** [comment]
Yes, there are already designs in JsSIP, but the expected state is inconsistent.

**Comment by cloudwebrtc** [comment]
I use this branch to test the call on FreeSwitch, answered after receiving the early media, it seems to work fine.

**Comment by rlsutton1** [comment]
It's a holiday here tomorrow, but I can arrange access to an asterisk system next week if that helps.

**Comment by cloudwebrtc** [comment]
@rlsutton1 If you can, please send the asterisk test account to my Github email to reproduce the error, but I have a week holiday and go online again on October 3.

**Comment by cloudwebrtc** [comment]
The expected state should be RTCSignalingStateStable because the SDP negotiation has been completed.

The explanation of `stable` is as follows:
```
There is no ongoing exchange of offer and answer underway. 
This may mean that the RTCPeerConnection object is new,
in which case both the localDescription and remoteDescription are null;
**it may also mean that negotiation is complete and a connection has been established.**
```

==> **it may also mean that negotiation is complete and a connection has been established.** <==

So I think this PR is correct, and when you test on Asterisk, the state is RTCSignalingState.RTCSignalingStateHaveLocalOffer, I guess your state is after the completion of CreateOffer and SetLocalDescription, but before receiving the 180/183 SDP, but I don't understand why I can't setRemoteDescription.

**Comment by mohbadar** [comment]
I’ve been attempting to connect with FreeSWITCH using the provided example in this repo, but without success. Can anyone provide me with a working example for connecting with FreeSWITCH? I’ve even tried using code generated by ChatGPT, but it didn’t work. I’ve been stuck on this for days and would appreciate any help.

**Pull request merged**